test/test_framework.py

```diff
@@ -499,83 +499,16 @@ class NoneEvent(EventBase):
 
 class TestStoredState(unittest.TestCase):
 
-    def setUp(self):
-        self.tmpdir = Path(tempfile.mkdtemp())
-
-    def tearDown(self):
-        shutil.rmtree(self.tmpdir)
-
-    def create_framework(self):
-        framework = Framework(self.tmpdir / "framework.data", self.tmpdir, None, None)
-        self.addCleanup(framework.close)
-        return framework
-
-    def test_basic_state_storage(self):
-        framework = self.create_framework()
-
-        class SomeObject(Object):
-            state = StoredState()
-
-        obj = SomeObject(framework, "1")
-
-        try:
-            obj.state.foo
-        except AttributeError as e:
-            self.assertEqual(str(e), "attribute 'foo' is not stored")
-        else:
-            self.fail("AttributeError not raised")
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
 
-        try:
-            obj.state.on = "nonono"
-        except AttributeError as e:
-            self.assertEqual(str(e), "attribute 'on' is reserved and cannot be set")
-        else:
-            self.fail("AttributeError not raised")
-
-        obj.state.foo = 41
-        obj.state.foo = 42
-        obj.state.bar = "s"
-
-        self.assertEqual(obj.state.foo, 42)
-
-        framework.commit()
-
-        # This won't be committed, and should not be seen.
-        obj.state.foo = 43
-
-        framework.close()
-
-        # Since this has the same absolute object handle, it will get its state back.
-        framework_copy = self.create_framework()
-        obj_copy = SomeObject(framework_copy, "1")
-        self.assertEqual(obj_copy.state.foo, 42)
-        self.assertEqual(obj_copy.state.bar, "s")
-
-    def test_mutable_types(self):
-        framework = self.create_framework()
-
-        class SomeObject(Object):
-            state = StoredState()
-
-        obj = SomeObject(framework, '1')
-        try:
-            class CustomObject:
-                pass
-            obj.state.foo = CustomObject()
-        except AttributeError as e:
-            self.assertEqual(str(e), 'attribute \'foo\' cannot be set to CustomObject: must be int/dict/list/etc')
-        else:
-            self.fail('AttributeError not raised')
-
-        framework.commit()
-
-        # Test and validation functions in a list of 2-tuples.
+        # Test and validation functions in a tuple of 2-tuples.
         # Assignment and keywords like del are not supported in lambdas so functions are used instead.
-        test_operations = [(
-            {},                                                            # Operand A.
-            None,                                                          # Operand B.
-            {},                                                            # Expected result.
-            lambda a, b: None,                                             # Operation to perform.
+        self.MUTABLE_TYPES_TEST_OPERATIONS = ((
```

Comment:
This feels like a worse approach than what we used to have previously. Now the overall type has the test for the individual cases, with a structure that only makes sense inside the individual cases. This PR was pretty much ready to be merged before, with only a trivial tuning to what was actually being modified.